internal/provider/deployment_resource.go

```diff
@@ -220,7 +220,7 @@ func (r *DeploymentResource) Create(ctx context.Context, req resource.CreateRequ
 	}
 
 	// Translate API response into Terraform state & save state
-	models.DeploymentAPIToResourceModel(updatedDeployment, &data)
+	data.FillFromAPIModel(updatedDeployment)
```

Comment:
I turned this into a method on `DeploymentResourceModel`, so now all the translation functions are methods on that type.